force-app/main/default/classes/DMLOperations.cls

```diff
@@ -148,6 +167,14 @@ public with sharing class DMLOperations {
         // Loop through each Opportunity name
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+        List<Opportunity> opp = new List<Opportunity>();
+        for(String oppName : oppNames)
+        {
+            Opportunity opp1 = new Opportunity(Name = oppName);
+            opp1.Account = [Select Id, Name FROM Account Where Name = :accountName];
+            opp.add(opp1);
+        }
+        insert opp;
```

Comment:
Your logic here is a little off. Review the question requirements again, it should only be creating Opportunities if the opp name is not already existing.